src/app/student/student-info.service.ts

```diff
@@ -0,0 +1,126 @@
+  }
+
+  getSpeciality(specId) {
+    return this.http.get(`Speciality/getRecords/${specId}`);
+  }
+
+  getTimeTableByGroup(groupId) {
+    return this.http.get(`timeTable/getTimeTablesForGroup/${groupId}`);
+  }
+
+  getSubjects() {
+    return this.http.get('Subject/getRecords');
+  }
+
+  getTests() {
+    return this.http.get('Test/getRecords');
+  }
+
+  getStudentInfo(id) {
+    const studentInfo: StudentInfo[] = [];
+    return this.getStudent(id).pipe(
+      switchMap((studentData: StudentInfo[]) => {
+        studentInfo.push(studentData[0]);
+        return this.getGroup(studentInfo[0].group_id);
+      }),
+      switchMap((groupData: Group[]) => {
+        studentInfo[0].group_name = groupData[0].group_name;
+        studentInfo[0].faculty_id = groupData[0].faculty_id;
+        studentInfo[0].speciality_id = groupData[0].speciality_id;
+        return this.getFaculty(studentInfo[0].faculty_id);
+      }),
+      switchMap((facultyData: Faculty[]) => {
+        studentInfo[0].faculty_name = facultyData[0].faculty_name;
+        return this.getSpeciality(studentInfo[0].speciality_id);
+      }),
+      switchMap((specData: Speciality[]) => {
+        studentInfo[0].speciality_name = specData[0].speciality_name;
+        studentInfo[0].speciality_code = specData[0].speciality_code;
+        return studentInfo;
+      })
+    );
+  }
+
+  getTimeTable(groupId) {
+    const timeTable = [];
+    return this.getTimeTableByGroup(groupId).pipe(
+      switchMap((timeTableData: TimeTable[]) => {
+        timeTableData.forEach(item => {
+          delete item.timetable_id;
+          delete item.group_id;
+        });
+        timeTable.push(timeTableData);
+        return this.getSubjects();
+      }),
+      switchMap((subjectsData: Subject[]) => {
+        timeTable[0].forEach(value => {
+          subjectsData.map(value1 => {
+            if (value1.subject_id === value.subject_id) {
+              value.subject_name = value1.subject_name;
+            }
+          });
+        });
+        return timeTable;
+      })
+    );
+  }
+
+  getTestsInfo(timeTableData: TimeTable[]) {
+    const timeTable = timeTableData;
+    const filteredTests: any[] = [];
+    return this.getTests().pipe(
+      concatMap((testData: Test[]) => {
+        filteredTests.push(timeTable.map(tt => testData.filter(test => test.subject_id === tt.subject_id)));
+        return filteredTests;
+      })
+    );
+  }
+
+  getData(studentId) {
+    const student = [];
+    const timeTable = [];
+    const tests = [];
+    return this.getStudentInfo(studentId).pipe(
+      switchMap(studentData => {
+        student.push(studentData);
+        return this.getTimeTable(studentData.group_id);
+      }),
+      switchMap(timetableData => {
+        timeTable.push(timetableData);
+        return this.getTestsInfo(timeTable[0]);
+      }),
+      switchMap((testsData: any) => {
+        const merged = [].concat.apply([], testsData);
```

Comment:
Here you want to push **testsData** into merged array? If it's true, i think we could use this: 
`const merged = [...testsData], or const merged.push(...testsData);`

Reply:
const merged = [...testsData] is preferable. And here is no need to use apply for context binding. [].concat means the same for the current case, all new data will be added to array which is created before dot. In case with apply, concat method will be use from array instance witch is created before the dot, but data will be added to instance witch is created inside apply brackets.